scripts/update-browser-releases/edge.ts

```diff
@@ -0,0 +1,271 @@
+/* This file is a part of @mdn/browser-compat-data
+ * See LICENSE file for more information. */
+
+import * as fs from 'node:fs';
+
+import chalk from 'chalk-template';
+
+import { newBrowserEntry, updateBrowserEntry } from './utils.js';
+
+/**
+ * getFutureReleaseDate - Read the future release date
+ * @param {string} release The release number of the versin
+ * @param {string} releaseScheduleURL The url of the MD file having it
+ * @returns {string} The date in the YYYY-MM-DD format
+ */
+const getFutureReleaseDate = async (release, releaseScheduleURL) => {
+  // Fetch the MD of the release schedule
+  const scheduleMD = await fetch(releaseScheduleURL);
+  const text = await scheduleMD.text();
+  if (!text) {
+    console.error(chalk`{red Release file not found.}`);
+    return '';
+  }
+  // Find the line
+  //const regexp = new RegExp(`| ${release} |\\w*|\\w*| ?Week of (\\w*) ?|\\w*|`, 'i');
+  const regexp = new RegExp(
+    `\\| ${release} \\|.*\\|.*\\| ?Week of (.*) ?\\|.*\\|`,
+    'i',
+  );
+  const result = text.match(regexp);
+  if (!result) {
+    console.log(chalk`{yellow Release date not found for Edge ${release}.}`);
+    return '';
+  }
+  const releaseDateText = result[1];
+
+  // Get the date frome the file
+  const months = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'May',
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+  const year = releaseDateText.substring(7, 11);
+  const month = months.indexOf(releaseDateText.substring(3, 6)) + 1;
+  const day = releaseDateText.substring(0, 2);
+
+  const releaseDate = new Date(`${year}-${month}-${day}Z`);
+
+  return releaseDate.toISOString().substring(0, 10); // Remove the time part
+};
+
+/**
+ * getReleaseNotesURL - Guess the URL of the release notes
+ * @param {string} status The status of the release
+ * @param {string} fullRelease The release of the release
+ * @param {string} date The date of the release
+ * @returns {string} The URL of the release notes or the empty string if not found
+ */
+const getReleaseNotesURL = async (status, fullRelease, date) => {
+  // If the status isn't stable, do not give back any release notes.
+  if (status !== 'Stable') {
+    return '';
+  }
+
+  const releaseStr = fullRelease.replace(/\./g, '');
+  const month = [
+    'january',
+    'february',
+    'march',
+    'april',
+    'may',
+    'june',
+    'july',
+    'august',
+    'september',
+    'october',
+    'november',
+    'december',
+  ];
+  const dateObj = new Date(date);
+  const dateStr = `${
+    month[dateObj.getMonth()]
+  }-${dateObj.getDate()}-${dateObj.getFullYear()}`;
+  return `https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel#version-${releaseStr}-${dateStr}`;
```

Comment:
This is probably more than enough to get started, but because this page is manually updated, there is a non-0 chance that the URL pattern changes. It's not going to break the script if it does, but the URLs in the edge.json file might become broken.
Do you think it's worth checking if this link works somehow?

Reply:
Yes, it is worth it. I've added a bit of code:
1. Normal case – release note page found AND section found:
![Capture d’écran 2023-10-25 à 07 12 29](https://github.com/mdn/browser-compat-data/assets/1466293/b6d63386-669d-4211-ac9b-00dfee2f3835)
2. Release note page not found
![Capture d’écran 2023-10-25 à 07 13 05](https://github.com/mdn/browser-compat-data/assets/1466293/de6904da-9ef0-47f6-97b9-95d0527fae31)
3. Section not found in the release note page
![Capture d’écran 2023-10-25 à 07 13 41](https://github.com/mdn/browser-compat-data/assets/1466293/4bedcd2e-bb2c-45c1-b3c0-8520640b7ba1)